c++/src/capnp/layout.c++

```diff
@@ -1609,31 +1636,61 @@ struct WireHelpers {
       return { segment, ptr };
     } else {
       // List of structs.
+      WordCount declDataSize = roundBitsUpToWords(value.structDataSize);
```

Comment:
Before this line can you add:

    KJ_DASSERT(value.structDataSize % BITS_PER_WORD == 0);

? I think it holds here (because `elementSize` wouldn't be INLINE_COMPOSITE for a list of sub-word structs) but I'm not 100% sure. If it doesn't hold there will be a unit test failure. (Make sure you're compiling in debug mode, i.e. without `-DNDEBUG`.)

Assuming it does hold then you can replace the `roundBitsUpToWords` with a simple division.

If it doesn't hold then you will have problems below, similar to the issues in `setStructPointer()`.